File: python/hardware/hil_get_string_property_example/hil_get_string_property_example.py
#################################################################
#
# hil_get_string_property_example.py - Python file
#
# This example gets the string propery (serial number) from a device.
#
# This example demonstrates the use of the following functions:
#    HIL.open
#    HIL.get_hil_string_property
#    HIL.close
#
# Copyright (C) 2023 Quanser Inc.
#################################################################

from quanser.hardware import HIL, HILError, StringProperty
import array as arr
import sys

board_type = "qube_servo2_usb"
board_identifier = "0"

argc = len(sys.argv)
if argc > 1:
    board_type = sys.argv[1]
    if argc > 2:
        board_identifier = sys.argv[2]

try:
    card = HIL()
    card.open(board_type, board_identifier)

    try:
        serial_number = card.get_string_property(StringProperty.SERIAL_NUMBER, 64)
        print("Serial number for %s: %s" % (board_type, serial_number))

    except HILError as ex:
        print("Unable to get serial number. %s" % ex.get_error_message())

    finally:
        card.close()

except HILError as ex:
    print("Unable to open board. %s" % ex.get_error_message())

input("Press Enter to continue.")
exit(0)

File: C/hardware/hil_get_string_property_example/hil_get_string_property_example.c
//////////////////////////////////////////////////////////////////
//
// hil_get_string_property_example.c - C file
//
// This example gets the string propery (serial number) from a device.
//
// This example demonstrates the use of the following functions:
//    hil_open
//    hil_get_hil_string_property
//    hil_close
//
// Copyright (C) 2021 Quanser Inc.
//////////////////////////////////////////////////////////////////
    
#include "hil_get_string_property_example.h"

int main(int argc, char* argv[])
{
    static const char  board_type[]       = "qube_servo2_usb";
    static const char  board_identifier[] = "0";
    static char        message[512];

    t_card board;
    t_int  result;

    if (argc > 1)
        if (argc > 2)
            result = hil_open(argv[1], argv[2], &board);
        else
            result = hil_open(argv[1], board_identifier, &board);
    else
        result = hil_open(board_type, board_identifier, &board);

    if (result == 0)
    {
        char serial_number[32];

        result = hil_get_string_property(board, PROPERTY_STRING_SERIAL_NUMBER, serial_number, 32);
        if (result >= 0)
        {
            printf("Serial number for %s: %s\n", board_type, serial_number);
        }
        else
        {
            msg_get_error_message(NULL, result, message, ARRAY_LENGTH(message));
            printf("Unable to read serial number. %s Error %d.\n", message, -result);
        }
            
        hil_close(board);
    }
    else
    {
        msg_get_error_message(NULL, result, message, ARRAY_LENGTH(message));
        printf("Unable to open board. %s Error %d.\n", message, -result);
    }

    return 0;
}
